self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2020, Self XDSD Contributors
+ * All rights reserved.
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to read the Software only. Permission is hereby NOT GRANTED to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software.
+ * <p>
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.core.projects;
+
+import com.jcabi.http.mock.MkAnswer;
+import com.jcabi.http.mock.MkContainer;
+import com.jcabi.http.mock.MkGrizzlyContainer;
+import com.selfxdsd.api.*;
+import com.selfxdsd.api.storage.Storage;
+import com.selfxdsd.core.RandomPort;
+import com.stripe.Stripe;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import javax.json.Json;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.HttpURLConnection;
+import java.time.LocalDateTime;
+
+/**
+ * Integration tests for {@link StripeWallet} payment.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.28
+ * @checkstyle ExecutableStatementCount (1000 lines).
+ */
+public final class StripeWalletITCase {
+
+    /**
+     * The rule for skipping test if there's BindException.
+     * @checkstyle VisibilityModifierCheck (3 lines)
+     */
+    @Rule
+    public final RandomPort resource = new RandomPort();
+
+    /**
+     * The StripeWallet successfully paying the Invoice.
+     * @throws IOException if something went wrong.
+     */
+    @Test
+    public void payingInvoiceOk() throws IOException {
+        try(
+            final MkContainer container = new MkGrizzlyContainer()
+                .next(
+                    new MkAnswer.Simple(
+                        HttpURLConnection.HTTP_OK,
+                        Json.createObjectBuilder()
+                            .add("id", "pi_1DiKBG2eZvKYlo2CRcpuIHvu")
+                            .add("created", 1545045758)
+                            .add("status", "succeeded")
+                            .build()
+                            .toString()
+                    )
+            ).start(this.resource.port())
```

Comment:
@criske Why are you starting a Mock server here? The Stripe API uses static methods, I don't see where you are configuring the URL.

Reply:
@amihaiemil Mock server url is configured at line 59 `Stripe.overrideApiBase(container.home().toString());`